src/jtag3.c

```diff
@@ -1597,6 +1600,27 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
       }
     }
 
+    else if (str_starts(extended_param, "mode=") &&
+      (str_starts(pgmid, "pickit4") || str_starts(pgmid, "snap"))) {
+      char mode[3];
```

Comment:
Should that not be `mode[4]` and `mode[3] = 0` set (otherwise how can `mode` ever be `str_caseeq()` with the *4*-character `"avr"`?